Editor/MenuActions/Selection/SelectMaterial.cs

```diff
@@ -78,20 +78,40 @@ public override ActionResult DoAction()
 
             UndoUtility.RecordSelection("Select Faces with Material");
 
-            HashSet<int> sel = new HashSet<int>(
-                    MeshSelection.topInternal
-                    .SelectMany(x => x.selectedFacesInternal.Select(y => y.submeshIndex)));
+            //Need to go from submesh index to material
+            List<Material> selectedMaterials = new List<Material>();
+            foreach(var pb in MeshSelection.topInternal)
+            {
+                HashSet<int> submeshIndex = new HashSet<int>(pb.selectedFacesInternal.Select(y => y.submeshIndex));
+                foreach(var index in submeshIndex)
+                {
+                    if(!selectedMaterials.Contains(pb.renderer.sharedMaterials[index]))
+                    {
+                        selectedMaterials.Add(pb.renderer.sharedMaterials[index]);
```

Comment:
Rather than creating two collections for submesh index and material, would it be more efficient to collect selected materials in a hashset?

Reply:
yes, make sense!